core/storage/topic/gae_models_test.py

```diff
@@ -0,0 +1,138 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test for Topic model."""
+
+from core.domain import subtopic_page_domain
+from core.domain import subtopic_page_services
+from core.domain import topic_domain
+from core.domain import topic_services
+from core.platform import models
+from core.tests import test_utils
+import feconf
+
+(topic_models,) = models.Registry.import_models([models.NAMES.topic])
+
+
+class TopicModelUnitTest(test_utils.GenericTestBase):
+    """Test the TopicModel class."""
+
+    def setUp(self):
+        super(TopicModelUnitTest, self).setUp()
+        self.topic = topic_domain.Topic.create_default_topic(
+            topic_id='topic_id',
+            name='topic_test'
+        )
+        topic_services.save_new_topic(feconf.SYSTEM_COMMITTER_ID, self.topic)
+
+    def test__trusted_commit(self):
+        # Compares topic model class variable (version)
+        # assigned in _trusted_commit method.
+        self.assertEqual(self.topic.version, 1)
+
+    def test_get_by_name(self):
+        self.assertEqual(
+            topic_models.TopicModel.get_by_name(self.topic.name).name,
+            self.topic.name
+        )
+        self.assertEqual(
+            topic_models.TopicModel.get_by_name(self.topic.name).id,
+            self.topic.id
+        )
+
+
+class TopicCommitLogEntryModelUnitTest(test_utils.GenericTestBase):
+    """Test the TopicCommitLogEntryModel class."""
+
+    def setUp(self):
+        super(TopicCommitLogEntryModelUnitTest, self).setUp()
+        self.topic = topic_domain.Topic.create_default_topic(
+            topic_id='topic_id',
+            name='topic_test'
+        )
+        topic_services.save_new_topic(feconf.SYSTEM_COMMITTER_ID, self.topic)
+        self.topic_test_instance_id = 'topic-%s-%s' % (
+            self.topic.id,
+            self.topic.version
+        )
+
+    def test__get_instance_id(self):
```

Comment:
Hi @tonadev I had a chat with @BenHenning about this. Here's what we both suggest: if you look at the superclass, the create() method is public, and makes use of _get_instance_id(). 

So one way you can test the code in this private method is to create a new commit log entry model, and introspect its ID to see if it's what you would expect. In particular you'll want to ensure that the topic ID and the version number are part of the commit ID. You could also then try retrieving the model and seeing if that works fine. (This is what I meant by "presumably some public method in that class uses the protected method behind-the-scenes; it is only the former that should be directly called in tests (but in a way that exercises the protected member)".)

Does this help? Please let us know if you need more clarification.

Also /cc @vibhor98 @apb7 for info -- this should be a general guideline for testing protected members.

Reply:
Done. 
Commit: https://github.com/oppia/oppia/pull/5689/commits/005859b796134baf31224776df9d8820901e1480